modules/gapi/include/opencv2/gapi/python_bridge.hpp

```diff
@@ -0,0 +1,177 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2019-2020 Intel Corporation
+
+#ifndef OPENCV_GAPI_PYTHON_BRIDGE_HPP
+#define OPENCV_GAPI_PYTHON_BRIDGE_HPP
+
+#include <opencv2/gapi/garg.hpp>
+#include <opencv2/gapi/gopaque.hpp>
+#include <iostream>
+
+namespace cv {
+namespace gapi {
+
+// NB: cv.gapi.CV_BOOL in python
+// As cv::detail::OpaqueKind except: CV_UNKNOWN, CV_DRAW_PRIM, CV_UINT64
+enum ArgType {
+    CV_BOOL,
+    CV_INT,
+    CV_DOUBLE,
+    CV_FLOAT,
+    CV_STRING,
+    CV_POINT,
+    CV_POINT2F,
+    CV_SIZE,
+    CV_RECT,
+    CV_SCALAR,
+    CV_MAT,
+};
+
+} // namespace gapi
+
+template <template <typename> class Wrapper, typename T>
+struct WrapType { using type = Wrapper<T>; };
+
+template <template <typename> class T, typename... Types>
+using MakeVariantType = cv::util::variant<typename WrapType<T, Types>::type...>;
+
+struct GAPI_EXPORTS_W_SIMPLE GOpaqueT
+{
+    using Storage = MakeVariantType<  cv::GOpaque
+                                    , bool
+                                    , int
+                                    , double
+                                    , float
+                                    , std::string
+                                    , cv::Point
+                                    , cv::Point2f
+                                    , cv::Size
+                                    , cv::Rect>;
+    template<typename T>
+    GOpaqueT(cv::GOpaque<T> opaque) : arg(opaque) { };
+
+    GAPI_WRAP GOpaqueT(gapi::ArgType type)
+    {
+
+#define HANDLE_CASE(T, K) case gapi::ArgType::CV_##T: \
+                arg = cv::GOpaque<K>(); \
+                break;
+
+        switch (type)
+        {
+            HANDLE_CASE(BOOL,    bool);
+            HANDLE_CASE(INT,     int);
+            HANDLE_CASE(DOUBLE,  double);
+            HANDLE_CASE(FLOAT,   float);
+            HANDLE_CASE(STRING,  std::string);
+            HANDLE_CASE(POINT,   cv::Point);
+            HANDLE_CASE(POINT2F, cv::Point2f);
+            HANDLE_CASE(SIZE,    cv::Size);
+            HANDLE_CASE(RECT,    cv::Rect);
+#undef HANDLE_CASE
+            default:
+                GAPI_Assert(false && "Unsupported type");
+        }
+    }
+
+    cv::detail::GOpaqueU strip() {
```

Comment:
Actually, I would rename that

Reply:
Why ? What's your suggestion ?

Reply:
I think it's a bit harsh word but maybe it's just me:) You could use `unpack` or `extract`

Reply:
Ok, don't think that it's really big effort to change that. Let's wait for the @dmatveev comments.